tests/test_modules/test_dataset_analysis/test_syntax_tagging.py

```diff
@@ -2,16 +2,18 @@
 # This source code is licensed under the Apache 2.0 license found in the LICENSE file
 # in the root directory of this source tree.
 from azimuth.modules.dataset_analysis.syntax_tagging import SyntaxTaggingModule
-from azimuth.types import DatasetSplitName
+from azimuth.types import DatasetColumn, DatasetSplitName
 from azimuth.types.tag import SmartTag
 
 
-def test_syntax_tagging(tiny_text_config):
-    mod = SyntaxTaggingModule(
-        DatasetSplitName.eval,
-        tiny_text_config,
-    )
+def verify_syntax_results(json_output, smart_tags_per_idx, token_count_per_idx):
+    for idx, res in enumerate(json_output):
+        assert all(v for k, v in res.tags.items() if k in smart_tags_per_idx[idx])
+        assert not any(v for k, v in res.tags.items() if k not in smart_tags_per_idx[idx])
```

Comment:
Would that be equivalent? The error would be a lot clearer, with the actual values and not just `False`.
```suggestion
        assert list(res.tags.values()) == [k in smart_tags_per_idx[idx] for k in res.tags.keys()]
```

Reply:
Looks equivalent to me, sounds good! Good idea for the error message.